src/test/java/de/unistuttgart/overworldbackend/CloneTest.java

```diff
@@ -0,0 +1,139 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.unistuttgart.overworldbackend.client.ChickenshockClient;
+import de.unistuttgart.overworldbackend.data.Course;
+import de.unistuttgart.overworldbackend.data.CourseDTO;
+import de.unistuttgart.overworldbackend.data.CourseInitialData;
+import de.unistuttgart.overworldbackend.data.mapper.CourseMapper;
+import de.unistuttgart.overworldbackend.repositories.CourseRepository;
+import java.io.File;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@AutoConfigureMockMvc
+@Testcontainers
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CloneTest {
+
+  @Container
+  public static DockerComposeContainer compose = new DockerComposeContainer(
+    new File("src/test/resources/docker-compose-test.yaml")
+  )
+    .withExposedService("overworld-db", 5432)
+    .withExposedService("reverse-proxy", 80);
+
+  @DynamicPropertySource
+  public static void properties(DynamicPropertyRegistry registry) {
+    registry.add(
+      "spring.datasource.url",
+      () ->
+        String.format(
+          "jdbc:postgresql://%s:%d/postgres",
+          compose.getServiceHost("overworld-db", 5432),
+          compose.getServicePort("overworld-db", 5432)
+        )
+    );
+    registry.add(
+      "chickenshock.url",
+      () -> String.format("http://%s/minigames/chickenshock/api/v1", compose.getServiceHost("reverse-proxy", 80))
+    );
+  }
+
+  @Autowired
+  MockMvc mvc;
+
+  @Autowired
+  private CourseRepository courseRepository;
+
+  @Autowired
+  private ChickenshockClient chickenshockClient;
+
+  @Autowired
+  private CourseMapper courseMapper;
+
+  private String fullURL;
+  private ObjectMapper objectMapper;
+
+  @BeforeEach
+  public void createBasicData() {
+    fullURL = "/courses";
+    objectMapper = new ObjectMapper();
+
+    //wait till chickenshock-backend is ready
+    boolean completed = false;
+    while (!completed) {
+      try {
+        chickenshockClient.getConfiguration(UUID.fromString("70fcd00c-b67c-46f2-be73-961dc0bc8de1"));
+        completed = true;
+      } catch (Exception e) {
+        completed = false;
+      }
```

Comment:
I think this is supposed to be a (misimplemented) circuit breaker that can actually deadlock the entire backend…

Reply:
Shouldn't the `FeignClient` already retry to get it?
We definitely need a **system to report minigame configurations where cloning failed**.
At the moment, it will simply copy until it encounters an error, which is not a good idea.
Also, the clones for Finitequiz and crosswordpuzzle instances are currently missing.

Reply:
Oh, just noticed you did half the work already.